store/sqlstore/file_info_store.go

```diff
@@ -753,3 +753,38 @@ func (fs SqlFileInfoStore) GetStorageUsage(allowFromCache, includeDeleted bool)
 	}
 	return size, nil
 }
+
+// GetUptoNSizeFileTime returns the CreateAt time of the last accessible file with a total size upto n bits.
+func (fs *SqlFileInfoStore) GetUptoNSizeFileTime(n int64) (int64, error) {
+	if n <= 0 {
+		return 0, errors.New("n can't be less than 1")
+	}
+
+	sizeSubQuery := sq.
+		Select("SUM(fi.Size) OVER(ORDER BY CreateAt DESC, fi.Id) RunningTotal", "fi.CreateAt").
```

Comment:
Does `SUM(fi.Size)` need to be wrapped in with a coalesce defaulting to 0, like [`GetStorageUsage`](https://github.com/mattermost/mattermost-server/pull/20454/files#diff-426ad80fd27e13823cb670cddf7ed901676bbe8982a238762b5400b7df6e67b6R742) did to avoid causing errors in the case no files have been uploaded in the instance?

Reply:
Not needed, as I am already handling that error.
```go
if err == sql.ErrNoRows {
  return 0, store.NewErrNotFound("File", "none")
}
```